### PR TITLE
Address buildout replacement #2432

### PR DESCRIPTION
**Pull request by phillxnet**
Entirely untested and incomplete -- save point.

Created to encourage community input on development in this direction re:
Fixing #2432 

Progress to date:
- Most of the "Required" fields with some: nice to haves-
- Plus the beginnings of grouped dependencies.
- 
- [x] Remove all buildout, bootstrap, zc.sourcerelease etc dependencies and special-case. Our proposed Poetry build system should replace all the associated processes as any transition period where one is doing some things and another some other (and possible by accident more of the same) will be way more confusing that a clean start on Poetry.
- [x] Remove djangorecipe (https://pypi.org/project/djangorecipe/) last updated **Djangorecipe: easy install of Django with buildout** Last updated to 2.2.1 Released: Jun 29, 2016.
- [x] Hard-wire hard-code all paths associated with our current buildout bootstrap steps. this can be un-done at a later date but should make for a speedier transition (new path to be existing "/opt/rockstor" as per existing rpm installs)
- [x] Hard-wire hard-code all paths associated with our conf pre-processors within buildout.
- [x] Establish /opt/rockstor/etc/nginx/mime.types (from conf).
- [x] Establish /opt/rockstor/etc/nginx/nginx.conf  (from conf).
- [ ] ~~Create Poetry plugin to download our jslibs.tar.gz and unzip into /opt/rockstor/static/js directory such that there is then an /opt/rockstor/static/js/dev directory containing all our backbone, bootstrap etc js libs, see: https://github.com/rockstor/rockstor-jslibs. N.B. this need be no more than running a curl tar de-archive command via a Poetry hook or the like.~~
- [x] The above is problematic given poetry's maturity and on-top of this our need, currently, to use an older version to empower us in our move from Python 2->3. Instead we can, for now, achieve our goals by carrying a build.sh script to assist in preparing our source prior to a poetry build.
- [x] Instantiate directly our supervisord.conf for our rockstor.service "/opt/rockstor/bin/supervisord -c /opt/rockstor/etc/supervisord.conf"
- [x] Instantiate a hardwired /opt/rockstor/etc/init.d/gunicorn
- [x] Run collectstatic during our poetry build, possibly via plugin.
- [ ] ~~It is proposed that we now replace our prior use of supervisord/supervisorctl to manage gunicorn, replicad, data-collector, "django run_huey" by migrating the same services to systemd scripts. And in the case of nginx, reverting to an override config of the os systemd.service which our prior buildbot normally disabled as we ran it from supervisord.~~
- [x] Avoid the overly large changes imposed by the above (for now) and just transition our nginx over from supervisord/ctrl and instead apply our config  via a systemd override/drop-file.
- [x] Replace our buildout based django-hack.py derived ./bin/django start script run, in turn, buy supervisord.
- [x] Fix an inadvertent bug re nginx systemd service likely being disabled left over form our prior management of that service from within supervisord.
- [x] fix nginx-stdout/stderr log locations re Logs Manager as we no longer run nginx under supervisord but via systemd as of this pull request.
- [x] Fix bug where we still, inadvertently, populate our new Poetry created .venv virtualenv directory with the old builout django-hack.py script (in initrock).

## Notes
- poetry-core=1.1.15 as this is the last Poetry version to supports 2.7
- We may well require some of our own simple Poetry plugins: https://python-poetry.org/docs/plugins/ which should help with our transition so we have like-for-like methods from our old buildout to the new poetry Orchestration.

### CODE REVIEW

**Comment by phillxnet** [comment]
The pyproject.toml file added by this pull request is a configuration file standard defined in PEP 518:
https://peps.python.org/pep-0518/

**Comment by phillxnet** [comment]
@FroggyFlox Thanks, I did see a load of new stuff adding in 1.2 but missed the dependency groups. Shame. we will have to pull them out: I started with a flat list and that will have to do us until we are further down this path. Cheers.

I was quite pleased to find that feature also! Never mind, mostly a nice to have for now anyway.
I should be back on this again tomorrow hopefully so I'll push that as a change first thing.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    "Philip Paul Guyton <support@linuxlines.com>",
+]
+license = "GPL-2.0-or-later"
+exclude = [
+    {path = "test"},
+]
+
+# We use groups to clarify our dependencies (essentially labels):
+# https://python-poetry.org/docs/managing-dependencies/#dependency-groups
+[tools.poetry.dependencies]
```
Comment (FroggyFlox) [inline pyproject.toml:30]: I believe this needs to be without the "s":
```suggestion
[tool.poetry.dependencies]
```

**Comment by FroggyFlox** [comment]
> I was quite pleased to find that feature also! Never mind, mostly a nice to have for now anyway.

Not a problem at all; we can simply comment those out so that we can just uncomment them once we're on Poetry 2 and we'll be set, I believe.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+psutil = "==5.9.4"
+mock = "==1.0.1"
+# pyzmq requires libzmq5 on system unless in wheel form.
+pyzmq = "== 19.0.2"  # Last specifying Python 2 on PyPi page.
+distro = "==1.6.0"  # Last Python 2/3 version that works as we expect.
+URLObject = "==2.1.1"
+
+# [tool.poetry.dev-dependencies]
+
+
```
Comment (FroggyFlox) [inline pyproject.toml:69]: Add scripts section:
```suggestion

[tool.poetry.scripts]
backup-plugin = 'backup.scheduler:main'
backup-config = 'scripts.config_backup:main'
bootstrap = 'scripts.bootstrap:main'
data-collector = 'smart_manager.data_collector:main'
debug-mode = 'scripts.debugmode:main'
delete-api-key = 'scripts.delete_api_key:main'
delete-rockon = 'scripts.rockon_delete:delete_rockon'
#docker-wrapper = 'scripts.docker_wrapper:main'
flash-optimize = 'scripts.flash_optimize:main'
initrock = 'scripts.initrock:main'
mnt-share = 'scripts.mount_share:mount_share'
ovpn-client-gen = 'scripts.ovpn_util:client_gen'
ovpn-client-print = 'scripts.ovpn_util:client_retrieve'
ovpn-initpki = 'scripts.ovpn_util:initpki'
prep_db = 'scripts.prep_db:main'
pwreset = 'scripts.pwreset:main'
qgroup-clean = 'scripts.qgroup_clean:main'
qgroup-maxout-limit = 'scripts.qgroup_maxout_limit:main'
qgroup-test = 'scripts.qgroup_test:main'
rcli = 'cli.rock_cli:main'
replicad = 'smart_manager.replication.listener_broker:main'
rockon-json = 'scripts.rockon_util:main'
send-replica = 'scripts.scheduled_tasks.send_replica:main'
st-pool-scrub = 'scripts.scheduled_tasks.pool_scrub:main'
st-snapshot = 'scripts.scheduled_tasks.snapshot:main'
st-system-power = 'scripts.scheduled_tasks.reboot_shutdown:main'
```

**Comment by phillxnet** [comment]
@FroggyFlox Thanks for the once over, I'll see if I can nudge this along a tad more.

**Comment by phillxnet** [comment]
Noting to reconstruct our supervisord.conf:
```
rleap15-3:~ # systemctl status rockstor
● rockstor.service - RockStor startup script
     Loaded: loaded (/etc/systemd/system/rockstor.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2022-11-24 16:34:40 GMT; 49min ago
   Main PID: 13494 (supervisord)
      Tasks: 13
     CGroup: /system.slice/rockstor.service
             ├─13494 /usr/bin/python2 /opt/rockstor/bin/supervisord -c /opt/rockstor/etc/supervisord.conf
             ├─13537 nginx: master process /usr/sbin/nginx -c /opt/rockstor/etc/nginx/nginx.conf
             ├─13538 /usr/bin/python2 /opt/rockstor/bin/gunicorn --bind=127.0.0.1:8000 --pid=/run/gunicorn.pid --workers=2 --log-file=/opt/rockstor/var/log/gunicorn.log --pythonpath=/opt/rockstor/src/rockstor --timeout=120 --graceful-timeout=120 wsgi:application
             ├─13539 /usr/bin/python2 /opt/rockstor/bin/data-collector
             ├─13540 /usr/bin/python2.7 /opt/rockstor/bin/django run_huey --workers 2 --worker-type thread --logfile /opt/rockstor/var/log/huey.log
             ├─13541 nginx: worker process
             ├─13542 nginx: worker process
             ├─13554 /usr/bin/python2 /opt/rockstor/bin/gunicorn --bind=127.0.0.1:8000 --pid=/run/gunicorn.pid --workers=2 --log-file=/opt/rockstor/var/log/gunicorn.log --pythonpath=/opt/rockstor/src/rockstor --timeout=120 --graceful-timeout=120 wsgi:application
             └─13560 /usr/bin/python2 /opt/rockstor/bin/gunicorn --bind=127.0.0.1:8000 --pid=/run/gunicorn.pid --workers=2 --log-file=/opt/rockstor/var/log/gunicorn.log --pythonpath=/opt/rockstor/src/rockstor --timeout=120 --graceful-timeout=120 wsgi:application

Nov 24 16:34:41 rleap15-3 supervisord[13494]: 2022-11-24 16:34:41,086 CRIT Server 'unix_http_server' running without any HTTP authentication checking
Nov 24 16:34:41 rleap15-3 supervisord[13494]: 2022-11-24 16:34:41,086 INFO supervisord started with pid 13494
Nov 24 16:34:42 rleap15-3 supervisord[13494]: 2022-11-24 16:34:42,089 INFO spawned: 'nginx' with pid 13537
Nov 24 16:34:42 rleap15-3 supervisord[13494]: 2022-11-24 16:34:42,091 INFO spawned: 'gunicorn' with pid 13538
Nov 24 16:34:42 rleap15-3 supervisord[13494]: 2022-11-24 16:34:42,093 INFO spawned: 'data-collector' with pid 13539
Nov 24 16:34:42 rleap15-3 supervisord[13494]: 2022-11-24 16:34:42,094 INFO spawned: 'ztask-daemon' with pid 13540
Nov 24 16:34:44 rleap15-3 supervisord[13494]: 2022-11-24 16:34:44,120 INFO success: data-collector entered RUNNING state, process has stayed up for > than 2 seconds (startsecs)
Nov 24 16:34:47 rleap15-3 supervisord[13494]: 2022-11-24 16:34:47,124 INFO success: nginx entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
Nov 24 16:34:47 rleap15-3 supervisord[13494]: 2022-11-24 16:34:47,125 INFO success: gunicorn entered RUNNING state, process has stayed up for > than 5 seconds (startsecs)
Nov 24 16:34:52 rleap15-3 supervisord[13494]: 2022-11-24 16:34:52,132 INFO success: ztask-daemon entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)

```

**Comment by phillxnet** [comment]
Notes on moving our initrock nginx initiated use of supervisorctl (python service manager) to regular systemd management.
```
rleap15-4:/opt/rockstor # cat /usr/lib/systemd/system/nginx.service | grep ExecStart
ExecStartPre=/usr/sbin/nginx -t
ExecStart=/usr/sbin/nginx -g "daemon off;"
```
We need to modify nginx's config to use our own, to serve our purposes. But as cleanly and transparent as possible.

Previously we disabled the os systemd service and managed the server via supervisord/supervisorctl.

**Comment by phillxnet** [comment]
Closing as:
Superseded by #2434

**Pull request closed without merging**